modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearOptimisticTxPrepareFuture.java

```diff
@@ -1059,6 +1059,12 @@ private void remap() {
         /** {@inheritDoc} */
         @SuppressWarnings("EqualsWhichDoesntCheckParameterClass")
         @Override public boolean equals(Object o) {
+            if (this == o)
+                return true;
+
+            if (!(o instanceof MappingKey))
```

Comment:
I think that this condition looks less readable than without 'NOT'.
Look at 'java.lang.String#equals'.

That touches all 'equals' methods.